pkg/channel/consolidated/reconciler/controller/kafkachannel.go

```diff
@@ -256,29 +250,12 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, kc *v1beta1.KafkaChannel
 	return newReconciledNormal(kc.Namespace, kc.Name)
 }
 
-func (r *Reconciler) setupSubscriptionStatusWatcher(ctx context.Context, channel *v1beta1.KafkaChannel) error {
-	var err error
-	groupIDPrefix := fmt.Sprintf("kafka.%s.%s", channel.Namespace, channel.Name)
-
-	m := func(cg string) bool {
-		return strings.HasPrefix(cg, groupIDPrefix)
-	}
-	err = r.consumerGroupWatcher.Watch(string(channel.ObjectMeta.UID), func() {
-		err := r.markSubscriptionReadiness(ctx, channel, r.consumerGroupWatcher.List(m))
-		if err != nil {
-			logging.FromContext(ctx).Errorw("error updating subscription readiness", zap.Error(err))
-		}
-	})
-	return err
-}
-
-func (r *Reconciler) markSubscriptionReadiness(ctx context.Context, ch *v1beta1.KafkaChannel, cgs []string) error {
+func (r *Reconciler) setupSubscriptionStatusWatcher(ctx context.Context, ch *v1beta1.KafkaChannel) error {
 	after := ch.DeepCopy()
 	after.Status.Subscribers = make([]v1.SubscriberStatus, 0)
 
 	for _, s := range ch.Spec.Subscribers {
-		cg := fmt.Sprintf("kafka.%s.%s.%s", ch.Namespace, ch.Name, s.UID)
-		if Find(cgs, cg) {
+		if r, _ := r.statusManager.IsReady(ctx, *ch, s); r {
```

Comment:
we've called IsReady but not used the error return value, is this the behaviour we're looking for?